src/multi_dict.jl

```diff
@@ -66,6 +67,46 @@ function in(pr::(Tuple{Any,Any}), d::MultiDict{K,V}) where {K,V}
     (v !== Base.secret_table_token) && (pr[2] in v)
 end
 
+# TODO: For efficiency, we probably want a MultiKeySet to correspond to Dict's KeySet
+
+# TODO: I'm not actually sure if we need this eachindex?
+function Base.eachindex(md::MultiDict)
+    flatten((repeated(k, length(vs)) for (k,vs) in md.d))
+end
```

Comment:
And this is the other TODO..

I don't remember what i was referring to... For Base.Dict, it looks like `eachindex` just calls `keys()`:
https://github.com/JuliaLang/julia/blob/7e8f2c05cd39aec8d4349ec915bf5c68cdf4c072/base/abstractarray.jl#L208-L209

So i'm not really sure why i wrote it?

--------

EDIT: AH, this is a similar question to the below. It's because in order for `collect()` to work, `eachindex()` has to return the same number of elements. So given `md = MultiDict(1=>2, 3=>3, 1=>4)`, `eachindex(md)` must return (unordered) `[1,1,3]`. But currently, `keys()` delegates to the dict:

```julia
julia> keys(md)
Base.KeySet for a Dict{Int64,Array{Int64,1}} with 2 entries. Keys:
  3
  1

julia> collect(eachindex(md))
3-element Array{Int64,1}:
 3
 1
 1
```

I guess i should probably just change this so that _keys_ returns `[3,1,1]`, and then just use the standard delegation from eachindex => keys. I think `keys()` should certainly include duplicates, because that's sort of the whole thing about a MultiDict.

So that needs to change too before this is finished.